### PR TITLE
Refactor GitHub Issues

### PR DESCRIPTION
**Pull request by mitchdenny**
This PR refactors the GitHub issues code base to be less of a command-line animal and more of an Azure Function with dependencies on AppConfiguration and KeyVault (injected as dependencies via a freshly configured DI container). This is to normalize it with the rest of the services we operate and build up the infrastructure needed to add some more capabilities.

### CODE REVIEW

**Comment by mitchdenny** [comment]
@AlexGhiondea just wanted to give you a heads up before I merge this in since its a bit of a departure from the last time you looked at the code. Secrets and config are now in KeyVault and AppConfig respectively.

**Comment by mitchdenny** [comment]
Also -- if the configuration of the ConfigurationClient looks odd its because we've got an inconsistency where you can't configure it via URI. I have this PR open to address that in the .NET repo:

https://github.com/Azure/azure-sdk-for-net/pull/13951

But the current approach will work until/if that gets merged.

File: tools/github-issues/Azure.Sdk.Tools.GitHubIssues/Reports/BaseReport.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        {
+            ConfigurationService = configurationService;
+        }
+
+        public async Task ExecuteAsync(ILogger log)
+        {
+            Task<IEnumerable<RepositoryConfiguration>> pendingRepositoryConfigurations = ConfigurationService.GetRepositoryConfigurationsAsync();
+            Task<string> pendingGitHubPersonalAccessToken = ConfigurationService.GetGitHubPersonalAccessTokenAsync();
+            Task<string> pendingSendGridToken = ConfigurationService.GetSendGridTokenAsync();
+            Task<string> pendingFromAddress = ConfigurationService.GetFromAddressAsync();
```
Comment (AlexGhiondea) [inline tools/github-issues/Azure.Sdk.Tools.GitHubIssues/Reports/BaseReport.cs:29]: I would move these inline. Some of those methods might have sync code that will now run at somewhat random times.

File: tools/github-issues/Azure.Sdk.Tools.GitHubIssues/Reports/ReportExecutionContext.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            this.SendGridToken = sendGridToken;
+            this.GitHubPersonalAccessToken = gitHubPersonalAccessToken;
+            this.RepositoryConfigurations = repositoryConfigurations;
+            this.GitHubClient = gitHubClient;
+        }
+
+        public ILogger Log { get; private set; }
+        public string FromAddress { get; private set; }
+        public string SendGridToken { get; private set; }
+        public string GitHubPersonalAccessToken { get; private set; }
```
Comment (AlexGhiondea) [inline tools/github-issues/Azure.Sdk.Tools.GitHubIssues/Reports/ReportExecutionContext.cs:26]: Do the tokens need to be available as properties here?

File: tools/github-issues/Azure.Sdk.Tools.GitHubIssues/Email/EmailSender.cs [hunk 3]
```diff
@@ -33,13 +33,7 @@ public static void SendEmail(string emailToken, string from, string template, st
             message.SetSubject($"GitHub Report: {title}");
             message.AddContent(MimeType.Html, template);
 
-
-#if !DEBUG
-            // Don't accidentally send email
             var emailResult = client.SendEmailAsync(message).GetAwaiter().GetResult();
-#else
-            System.IO.File.WriteAllText("output.html", template);
-#endif
```
Comment (AlexGhiondea) [inline tools/github-issues/Azure.Sdk.Tools.GitHubIssues/Email/EmailSender.cs:42]: Is there a good way now to run this locally and not worry about accidentally sending a ton of email?

Comment (mitchdenny) [reply tools/github-issues/Azure.Sdk.Tools.GitHubIssues/Email/EmailSender.cs:42]: I can add this back in, although when I'm debugging services what I typically do is run against the staging configuration environment which only has configuration to send e-mail to me.

File: tools/github-issues/Azure.Sdk.Tools.GitHubIssues/Reports/ReportExecutionContext.cs (hunk 2 above)
Comment (mitchdenny) [reply tools/github-issues/Azure.Sdk.Tools.GitHubIssues/Reports/ReportExecutionContext.cs:26]: Without them each report would need to take a dependency on SecretClient and query it directly. This seemed like a cleaner way to provide those values to each report.

File: tools/github-issues/Azure.Sdk.Tools.GitHubIssues/Email/EmailSender.cs (hunk 3 above)
Comment (mitchdenny) [reply tools/github-issues/Azure.Sdk.Tools.GitHubIssues/Email/EmailSender.cs:42]: Put this back in.

**Pull request merged**